### PR TITLE
Add support for OSPF summary

### LINKED ISSUE #634: Add support for route summary under router ospf
**Issue Type**

- [x] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

Update existing OSPF data model to support route summary in EOS CLI CONFIG GEN

**Describe the solution you'd like**

Update datamodel to add summary section

```yaml
router_ospf:
  process_ids:
    < process_id >:
      vrf: < vrf_name_for_process_id >
      passive_interface_default: < true | false >
      router_id: < IPv4_address >
      log_adjacency_changes_detail: < true | false >
      bfd_enable: < true | false >
      no_passive_interfaces:
        - < interface_1 >
        - < interface_2 >
      max_lsa: < integer >
      default_information_originate:
        always: true
      summary:
        - prefix: < summary_prefix_01 >
          tag: < string >
          attribute_map: < string >
          not_advertise: < true | false >
        - prefix: < summary_prefix_02 >
        - prefix: < summary_prefix_03 >
      redistribute:
        static:
          route_map: < route_map_name >
        connected:
          route_map: < route_map_name >

```

**Additional context**

Required for __`FR-PROJECT-02`__

### PR DESCRIPTION
**Pull request by titom73**
Add support for the following data model:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes issue #634 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

New data model implemented

```yaml
router_ospf:
  process_ids:
    < process_id >:
      summary:
        - prefix: < summary_prefix_01 >
          tag: < string >
          attribute_map: < string >
          not_advertise: < true >
        - prefix: < summary_prefix_02 >
        - prefix: < summary_prefix_03 >
```

## How to test

Check Molecule scenario

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by xaviramon** [comment]
Maybe not limited to OSPF summary itself, but the entire section does not show in which VRF the process is configured in device documentation. Something to take in mind in this PR or open a new issue?
Otherwise, LGTM!

**Comment by ClausHolbechArista** [comment]
rebasing, fixing conflicts and modifying a bit to match the rest of ospf docs (new style)

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml [hunk 1]
```diff
@@ -17,6 +17,22 @@ router_ospf:
...
       mpls_ldp_sync_default: true
+    101:
+      vrf: CUSTOMER01
+      passive_interface_default: true
+      router_id: 1.0.1.1
+      log_adjacency_changes_detail: true
+      bfd_enable: false
+      no_passive_interfaces:
+        - Ethernet2.101
+      summary:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml]: I would suggest to change to `summary_address`, to match the cli.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -1773,6 +1773,15 @@ router_ospf:
       max_lsa: < integer >
       default_information_originate:
         always: true
+      summary:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: change to `summary_address` to match cli.

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
please see comments!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed by commit 8b7c7977

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml]: Fixed by commit 8b7c7977

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**